Fix expected validation errors for `elem.drop`

fitzgen opened the pull request:
The validation for `elem.drop` only checks that the referenced segment is valid within the context, it does not check the existence of tables. Therefore, when asserting that a module is invalid because of a bad segment index in `elem.drop`, we should expect an error message that references an unknown element segment, not a message that references an unknown table.

binji commented:
These tests are run in the reference interpreter, where they currently pass. It looks like the issue is the [ElemDrop instruction validation](https://github.com/WebAssembly/bulk-memory-operations/blob/master/interpreter/valid/valid.ml#L264), which checks for the existence of table 0. Do  you mind fixing this (and `data.drop` too, which has the same issue)?

fitzgen commented:
Okay, I fixed the spec interpreter's validation and also updated the expected error messages for invalid `data.drop`s in `memory_init.wast`.

Let's definitely wait on CI since I did this all in the online github editor :)

fitzgen commented:
Okay, CI is finally green :)
